### PR TITLE
Improvement/vltclt 30 add typing

### PR DESCRIPTION
**Pull request by KazToozs**
This PR adds a type to VaultClient for TypeScript projects that call it.
Related to https://github.com/scality/Zenko/pull/1818#discussion_r1142458487

Not sure if this is the right approach to exporting, input on this is welcome
https://github.com/scality/vaultclient/pull/424/commits/22d1041675c339c9ef1e14b335c98b8c96a78b28

### CODE REVIEW

File: dist/IAMClient.d.ts
Comment (ghost) [inline dist/IAMClient.d.ts:1]: I don't think it should be in a dist folder, d.ts files should be in the same folder (with the same name) as the .js files

**Review by ghost — APPROVED** [review]

File: lib/IAMClient.d.ts [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+     *                       request has been processed
+     * @param {object} log - werelogs request logger
+     * @param {VaultClient~requestCallback} cb - callback
+     * @returns {undefined}
+     */
+    handleResponse(res: object, ret: string, log: object, cb: any): undefined;
+}
+import { http as HttpAgent } from "httpagent";
+import { https as HttpsAgent } from "httpagent";
+//# sourceMappingURL=IAMClient.d.ts.map
```
Comment (XinLi0207) [inline lib/IAMClient.d.ts:389]: can we delete this comment line or do we have some specific reason to keep it here?

Comment (KazToozs) [reply lib/IAMClient.d.ts:389]: honestly no idea as this was a generated file!

**Review by XinLi0207 — APPROVED** [review]

**Pull request merged**